app/services/proofing/resolution_result_adjudicator.rb

```diff
@@ -0,0 +1,46 @@
+module Proofing
+  class ResolutionResultAdjudicator
+    attr_reader :resolution_result, :state_id_result, :should_proof_state_id
+
+    def initialize(resolution_result:, state_id_result:, should_proof_state_id:)
+      @resolution_result = resolution_result
+      @state_id_result = state_id_result
+      @should_proof_state_id = should_proof_state_id
+    end
+
+    def adjudicated_result
+      FormResponse.new(
+        success: success?,
+        errors: resolution_result.errors.merge(state_id_result.errors),
+        extra: {
+          exception: resolution_result.exception || state_id_result.exception,
+          timed_out: resolution_result.timed_out? || state_id_result.timed_out?,
+          context: {
+            should_proof_state_id: should_proof_state_id,
+            stages: {
+              resolution: resolution_result.to_h,
+              state_id: state_id_result.to_h,
+            },
+          },
+        },
+      )
+    end
+
+    private
+
+    def success?
+      return true if resolution_result.success? && state_id_result.success?
+      return false unless should_proof_state_id
+      return true if state_id_attributes_cover_resolution_failures?
+      false
+    end
```

Comment:
first off, love this class
second, what if we added an internal reason slug or something to help us debug failures in the future?

like:
```suggestion
    # @return [Array(Boolean, Symbol)] a pair of (success, reason)
    def result_reason
      return [true, :pass_instant_verify_and_state_id] if resolution_result.success? && state_id_result.success?
      return [false, :skip_state_id] unless should_proof_state_id # TODO: better name?
      return [true, :state_id_attributes_cover_resolution_failures] if state_id_attributes_cover_resolution_failures?
      [false, :missing_data]
    end
```

Reply:
and then to be clear, we'd log the symbol somewhere in our response